### PR TITLE
Updating google analytics metrics getter

### PR DESCRIPTION
**Pull request by cansavvy**
#### What changes are being implemented in this Pull Request?

Fixing up some things for usability and DRYness purposes. 

The google analytics `all_ga_metrics()` is being deprecated changed so it can take property ids instead of just taking all properties from an account. 

In order to do this, we needed to make a new function `get_ga_property_info()` that will allow a display name (and other info) for a property to be retrieved.

### CODE REVIEW

**Comment by cansavvy** [comment]
I also ran styler::style_dir("R") so a lot of the changes have to do with that.

File: R/google-analytics.R [hunk 1]
```diff
@@ -295,14 +332,26 @@ link_clicks <- function() {
...
 #' accounts <- get_ga_user()
 #'
-#' some_stats_list <- get_all_ga_metrics(property_ids = property_ids)
+#' all_properties <- get_multiple_ga_metrics(account_id = accounts$id)
+#'
+#' some_properties <- get_multiple_ga_metrics(property_ids = c("422671031", "422558989"))
+#'
 #' }
-get_all_ga_metrics <- function(account_id = NULL, token = NULL, dataformat = "dataframe") {
+get_multiple_ga_metrics <- function(account_id = NULL, property_ids = NULL, token = NULL, dataformat = "dataframe",
```
Comment (cansavvy) [inline R/google-analytics.R:340]: And this function got reworked. Can you take a look and see what you think of the changes and if you have recommendations? @howardbaek

File: R/google-analytics.R [hunk 2]
```diff
@@ -127,6 +127,41 @@ get_ga_properties <- function(account_id, token = NULL) {
   return(results$properties)
 }
 
+#' Get all property information for a particular property id
```
[the remark this answers is not in the document]
Comment (howardbaek) [reply R/google-analytics.R:128]: This isn't working for me. As documented, I made sure to check the box `See and download your Google Analytics data.` on the OAuth screen:

Screenshot 2024-03-01 at 11 36 49 AM

However, when I see

```
> property_info <- get_ga_property_info(property_id = "422558989")
Using user-supplied cached tokens stored using authorize("google")


Error in request_ga(token = token, url = paste0("https://analyticsadmin.googleapis.com/v1alpha/properties/",  : 
  API request failed:{
  "error": {
    "code": 403,
    "message": "The caller does not have permission",
    "status": "PERMISSION_DENIED"
  }
}
```

File: R/google-analytics.R (hunk 1 above)
Comment (howardbaek) [reply R/google-analytics.R:338]: Same as above:

```
> some_properties <- get_multiple_ga_metrics(property_ids = c("422671031", "422558989"))
Using user-supplied cached tokens stored using authorize("google")


Error in request_ga(token = token, url = paste0("https://analyticsadmin.googleapis.com/v1alpha/properties/",  : 
  API request failed:{
  "error": {
    "code": 403,
    "message": "The caller does not have permission",
    "status": "PERMISSION_DENIED"
  }
}
```

**Review by howardbaek — CHANGES_REQUESTED** [review]
I think we may have to address the Google scopes as the functions here are denying permission.

File: R/google-analytics.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (cansavvy) [reply R/google-analytics.R:128]: @howardbaek Did you login with ITCR Gmail or your own personal gmail? If you logged in with your own gmail, you don't have access to this property id so it wouldn't work.

[the remark this answers is not in the document]
Comment (cansavvy) [reply R/google-analytics.R:128]: I should make that more clear in the example

**Comment by cansavvy** [comment]
We tested this in a pair programming session and found out it was because Howard was trying to access a property id with the hutchgmail account that it does not have access to. Tried to fix the docs appropriately.

**Pull request merged**